scalanet/src/io/iohk/scalanet/peergroup/TCPPeerGroup.scala

```diff
@@ -116,6 +123,10 @@ object TCPPeerGroup {
 
     override def sendMessage(message: M): Task[Unit] = {
       toTask(nettyChannel.writeAndFlush(Unpooled.wrappedBuffer(codec.encode(message)(bi))))
+        .onErrorRecoverWith {
+          case e: IOException =>
+            Task(throw new ChannelBrokenException[InetMultiAddress](to, e))
```

Comment:
I think you could call Task.raiseError and avoid throwing the exception

Reply:
Agreed.